src/tools/sc-hsm-tool.c

```diff
@@ -803,10 +821,9 @@ static void ask_for_password(char **pwd, int *pwdlen)
 
 
 
-static int generate_pwd_shares(sc_card_t *card, char **pwd, int *pwdlen, int password_shares_threshold, unsigned int password_shares_total)
+static int generate_pwd_shares(sc_card_t *card, char **pwd, int *pwdlen, int password_shares_threshold, int password_shares_total)
```

Comment:
If `password_shares_threshold` and `password_shares_total` must both be bigger than 2 (as tested below) then why do you bother with negative values here at all? Why don't you catch `-1` (and any other negative value) in `create_dkek_share`, which leaves `password_shares_threshold` with accepting `unsigned int`s only.